### DISCUSSION TITLE
esp-hosted vs ninaw10?

### DISCUSSION THREAD

**Discussion by victorallume** [micropython/micropython#14229] [Libraries & Drivers] [answered]
Do I understand correctly that both esp-hosted and ninaw10 fulfil the same function, namely to use an ESP32 device as a wifi (and optionally bluetooth) interface (for another MCU to use; not running micropython on the ESP32). Is there any reason to use one over the other?

**Comment by robert-hh** [comment, marked as answer]
Both NINAW10 and esp-hosted allow to use an ESP32 device as WiFi and BLE interface. There are subtle differences:
```
                NINAW10        esp-hosted
                NINAW10        esp-hosted
esp-idf version   3.3.         up to   5.x
of the module

Support ESP32     ESP32        ESP32, ESP32Sx
module                           ESP32C3
Supported by     Arduino        espressif

Location of the  Module         Host MCU
IP stack                         (LWIP)

Memory usage     Medium           Large
caused by LWIP

BLE test suite    fair            good
coverage
```
The WiFi speed is similar. The major differences are the type of support ESP32 module, host memory footprint and  BLE support quality,
